### DISCUSSION TITLE
More sub upgrade options: Installation Points

### DISCUSSION THREAD

**Discussion by leonard84** [FakeFishGames/Barotrauma#8632] [Ideas and suggestions]
## Installation Points
Currently, you can only upgrade your sub incrementally, i.e., get x% more performance or such per upgrade. The only exceptions are hardpoints for turrets. It would be really cool if a sub designer could create hardpoint-like placeholders inside the sub for installations, e.g. fabricators/deconstructors/linked storage/research station and so on. This way you could design a cheap sub, that has limited capabilities from the start, but could be upgraded to add more and more during the campaign. This makes for interesting choices on which installations to get vs. other purchases.
As all the things have different size, I don't propose to have a general purpose placeholder, but one for each installation, having the same connectors as the finished installation, so that it can be wired in the editor. Ideally, this could be generalized so that you can apply it to all installations.

**Comment by mLuby** [comment]
> As all the things have different size, I don't propose to have a general purpose placeholder

I _do_ suggest these [Utiility Hardpoints #6180](https://github.com/Regalis11/Barotrauma/discussions/6180) have a standard size (the Unit Load Device is a good size [as seen here](https://github.com/Regalis11/Barotrauma/discussions/6180#discussioncomment-1607557)) and have a standard connection to a relay/JB for power and data input/output. That way the community can design new "upgrades" to install without having to do a bunch of integration work per submarine.

**Reply by leonard84** [reply]
I think both suggestions have value, the downside of a uniform placeholder is that as a designer you have to work in those blocks and it might lead to a more uniform sub design with less individualism aesthetically. Also you can't have smaller optional upgrades, e.g. just an additional status monitor next in engineering.
My proposal would allow for a more fine-grained and individualistic design and upgrade experience. The utility hardpoints would go more for interchangeable parts.
IMO both proposals can be implemented without interfering with each other.

**Reply by mLuby** [reply]
That's a fair point. On the off-chance you weren't aware, [mods making installations placeable](https://steamcommunity.com/sharedfiles/filedetails/?id=2346495901) do exist (though yes these are suggestions for features to include in the _base_ game); those mods would let you add that status monitor mid-round.
